requests/models.py

```diff
@@ -447,6 +451,18 @@ def prepare_body(self, data, files, json=None):
         if is_stream:
             body = data
 
+            if getattr(body, 'tell', None) is not None:
+                # Record the current file position before reading.
+                # This will allow us to rewind a file in the event
+                # of a redirect.
+                try:
+                    # cast value to int for Python 2.7 which returns a long
+                    self._body_position = int(body.tell())
```

Comment:
The fact that Python 2.7 returns a long is, I suspect, meaningful. In particular, Python's 2.7's `int` is a 32/64-bit number, whereas the long is unbounded (it's a bignum).

For this reason, we should probably drop the cast and instead define an `integer_types` field in `compat.py` that is `(int, long)` on Python 2 and `int` on Python 3.